buildSrc/src/main/kotlin/io/spine/internal/gradle/report/license/ModuleDataExtensions.kt

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2021, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.internal.gradle.report.license
+
+import com.github.jk1.license.ModuleData
+import io.spine.internal.markup.MarkdownDocument
+import kotlin.reflect.KCallable
+
+/**
+ * This file declares the Kotlin extensions for `ModuleData`.
+ */
+
+/**
+ * Prints several of the module data dependencies under the section with the passed [title].
+ */
+internal fun MarkdownDocument.printSection(
+    title: String,
+    modules: Iterable<ModuleData>
+): MarkdownDocument {
+    this.h2(title)
+    modules.forEach {
+        it.print(this)
+    }
+    return this
+}
+
+/**
+ * Prints the metadata of the module to the specified [Markdown document][out].
+ */
+private fun ModuleData.print(out: MarkdownDocument) {
```

Comment:
As long as `out` parameter is common for most of the fun here, can we transform them to extension functions of `Markdown` document? Assuming that we start doing this in `printSelection()` it would look consistent and easier to understand.